web/src/features/fba/components/infoPanel/AdvisoryText.tsx

```diff
@@ -1,39 +1,99 @@
 import { Box, Typography } from '@mui/material'
-import { FireCenter, FireShapeAreaDetail } from 'api/fbaAPI'
+import { FireCenter, FireShape, FireZoneFuelStats } from 'api/fbaAPI'
 import { DateTime } from 'luxon'
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import { useSelector } from 'react-redux'
 import { selectProvincialSummary } from 'features/fba/slices/provincialSummarySlice'
+import { selectFireCentreHFIFuelStats } from '@/app/rootReducer'
 import { AdvisoryStatus } from 'utils/constants'
-import { groupBy } from 'lodash'
+import { isEmpty, isUndefined } from 'lodash'
 import { calculateStatusText } from '@/features/fba/calculateZoneStatus'
 
 interface AdvisoryTextProps {
   issueDate: DateTime | null
   forDate: DateTime
   selectedFireCenter?: FireCenter
   advisoryThreshold: number
+  selectedFireZoneUnit?: FireShape
 }
 
-const AdvisoryText = ({ issueDate, forDate, advisoryThreshold, selectedFireCenter }: AdvisoryTextProps) => {
+const AdvisoryText = ({
+  issueDate,
+  forDate,
+  selectedFireCenter,
+  advisoryThreshold,
+  selectedFireZoneUnit
+}: AdvisoryTextProps) => {
   const provincialSummary = useSelector(selectProvincialSummary)
+  const { fireCentreHFIFuelStats } = useSelector(selectFireCentreHFIFuelStats)
+  const [selectedFireZoneUnitTopFuels, setSelectedFireZoneUnitTopFuels] = useState<FireZoneFuelStats[]>([])
 
-  const getZoneStatusMap = (fireZoneUnitDetails: Record<string, FireShapeAreaDetail[]>) => {
-    const zoneStatusMap: Record<AdvisoryStatus, string[]> = {
-      [AdvisoryStatus.ADVISORY]: [],
-      [AdvisoryStatus.WARNING]: []
+  const [minStartTime, setMinStartTime] = useState<number | undefined>(undefined)
+  const [maxEndTime, setMaxEndTime] = useState<number | undefined>(undefined)
+
+  const sortByArea = (a: FireZoneFuelStats, b: FireZoneFuelStats) => {
+    if (a.area > b.area) {
+      return -1
+    }
+    if (a.area < b.area) {
+      return 1
     }
+    return 0
+  }
 
-    for (const zoneUnit in fireZoneUnitDetails) {
-      const fireShapeAreaDetails: FireShapeAreaDetail[] = fireZoneUnitDetails[zoneUnit]
-      const status = calculateStatusText(fireShapeAreaDetails, advisoryThreshold)
+  useEffect(() => {
+    if (
+      isUndefined(fireCentreHFIFuelStats) ||
+      isEmpty(fireCentreHFIFuelStats) ||
+      isUndefined(selectedFireCenter) ||
+      isUndefined(selectedFireZoneUnit)
+    ) {
+      setSelectedFireZoneUnitTopFuels([])
+      setMinStartTime(undefined)
+      setMaxEndTime(undefined)
+      return
+    }
+    const allZoneUnitFuelStats = fireCentreHFIFuelStats?.[selectedFireCenter.name]
+    const selectedZoneUnitFuelStats = allZoneUnitFuelStats?.[selectedFireZoneUnit.fire_shape_id] ?? []
+    const sortedFuelStats = [...selectedZoneUnitFuelStats].sort(sortByArea)
+    let topFuels: FireZoneFuelStats[] = []
+    for (let i = 0; i < 3 && i < sortedFuelStats.length; i++) {
```

Comment:
Could simplify with `take(sortedFuelStats, 3)`

https://lodash.com/docs/4.17.15#take

Reply:
Thanks for the tip! Changed in https://github.com/bcgov/wps/pull/3918/commits/26fbecdf625ce33c0fc99a4d700823ce92f6a1ce